### PR TITLE
[WIP] Video Annotation on CVAT-MONAI Label with Active Learning

### PR DESCRIPTION
**Pull request by tangy5**
The draft PR shows the development progress, not ready for review yet. 

The draft PR adds the video frame loader by cv2 and transforms. 
Added average strategy for video scoring. Select top K videos and push to CVAT server for annotation.

ToolTracking model is used for sample implementation.

Next step:

After done video annotation in CVAT, trigger video frame training and second round scoring for active learning.

### CODE REVIEW

**Comment by tangy5** [comment]
Current progress:

**Feature created and tested:** MONAI Label can directly load video studies, *.mp4, *.avi, *.mov, etc...LoadVideoFrame is created and send video frame for scoring module, then a video based metric is used to select top K video studies for labeling. 
The selected videos can be pushed to CVAT for labeling, MONAI Label server waiting for task completion, then retrieve labels for training. Repeat active learning process....

TODO and investigation: 
The video upload to CVAT is not working well, currently, it can creates each frame as a job. 
Need to investigate Rest API command: request post for create task:  each job with the video's all frames. 

E.g., 
 "  r = requests.post(f"{self.api_url}/api/tasks/{task_id}/data", files=files_, data=data, auth=self.auth).json()
"

```
        data = {
            "image_quality": 70,
            "client_files[0]": open(image, "rb"),
            "client_files":[
                {
                    "file": "66.mp4"
                }
            ],
            "use_zip_chunks": True,
            'chunk_size': 31, 
            'size': 0,
            'start_frame': 0, 
            'stop_frame': 31, 
            'compressed_chunk_type': 'video', 
            'original_chunk_type': 'video', 
            'client_files': ['66.mp4'], 
            'server_files': [], 
            'remote_files': [], 
            'use_cache': False, 
            'copy_data': False, 
            'storage_method': 'file_system', 
            'storage': 'local', 
            'sorting_method': 'lexicographical'
        }
```

File: monailabel/transform/pre.py [hunk 1]
```diff
@@ -60,3 +93,129 @@ def __call__(self, data):
...
+        return self
+
+    def __exit__(self, *_):
+        sys.stderr = self.old_stderr
+        os.dup2(self.old_stderr_fileno, self.old_stderr_fileno_undup)
+        os.close(self.old_stderr_fileno)
+        self.errnull_file.close()
+
+
+class LoadVideoFrame(Transform):
```
Comment (Nic-Ma) [inline monailabel/transform/pre.py:117]: May I know why you make it as a transform?
Maybe you can use the MONAI `VideoFileDataset` direclty:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/video_dataset.py#L141

Thanks.

File: monailabel/transform/pre.py [hunk 2]
```diff
@@ -60,3 +93,129 @@ def __call__(self, data):
             label[label > 0] = self.value
             d[key].array = label
         return d
+
+
+class SuppressStderr:
```
Comment (Nic-Ma) [inline monailabel/transform/pre.py:98]: MONAI already has this class:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/video_dataset.py#L33

Thanks.

**Review by Nic-Ma — COMMENTED** [review]
CC @SachidanandAlle for more discussion and comments.

Thanks in advance.

**Pull request closed without merging**